brave/brave-core
# [ads] Do not discard API calls during ads initialization

Resolves https://github.com/brave/brave-browser/issues/41628



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

Test case 1
- Start the Browser
- Do not join Brave Rewards
- Open search.brave.software
- Search with `giraffe` keyword
- Make sure that search result ad was shown
- Click on the search result ad
- EXPECTATION: Search result ad clicked event is triggered. There should be `Brave Ads` log:
   * Clicked search result ad with placement id %%% and creative instance id %%%

Test case 2
- Start the Browser
- Do not join Brave Rewards
- Join Brave News
- Scroll News Feed until `Brave news Ad` is shown
- EXPECTATION: Inline content ad impression event is triggered. There should be `Brave Ads` log:
   * Viewed inline content ad impression with placement id %%% and creative instance id %%%
- Click on the `Brave news ad`
- EXPECTATION: Inline content ad clicked event is triggered. There should be `Brave Ads` log:
   * Clicked inline content ad with placement id %%% and creative instance id %%%

Test case 3
- Start the Browser
- Join Brave Rewards
- Open search.brave.software
- Search with `giraffe` keyword
- Make sure that search result ad was shown
- EXPECTATION: Search result ad impression event is triggered. There should be `Brave Ads` log:
   * Viewed search result ad impression with placement id %%% and creative instance id %%%
- Click on the search result ad
- EXPECTATION: Search result ad clicked event is triggered. There should be `Brave Ads` log:
   * Clicked search result ad with placement id %%% and creative instance id %%%

Test case 4
- Start the Browser
- Join Brave Rewards
- Open NTP until NTT Sponsored Image ad is shown
- EXPECTATION: New tab page ad impression event is triggered. There should be `Brave Ads` log:
   * Viewed new tab page ad impression with placement id %%% and creative instance id %%%
- Click on the NTT Sponsored Image ad
- EXPECTATION: New tab page ad clicked event is triggered. There should be `Brave Ads` log:
   * Clicked new tab page ad with placement id %%% and creative instance id %%%

Test case 5
- Start the Browser
- Join Brave Rewards
- Trigger a push notification ad
- EXPECTATION: Notification ad impression event is triggered. There should be `Brave Ads` log:
   * Viewed notification ad impression with placement id %%% and creative instance id %%%
- Click on the push notification ad
- EXPECTATION: Notification ad clicked event is triggered. There should be `Brave Ads` log:
   * Clicked notification ad with placement id %%% and creative instance id %%%

Test case 6
- Start the Browser
- Join Brave Rewards
- Trigger any Brave Ad
- Open brave://rewards page
- EXPECTATION:
       * `Total ads received this month` has correct count for triggered Brave Ad
- Click `30-day Ads history`
- EXPECTATION:
       * Triggered ad is listed on `Ads History` dialog
- Toggle Like/Dislike/Save for an Ad and for the Ad Category
- Make sure that Likes/Dislikes/Save are stored after `Ads History` dialog is reopened

Test case 7
- Start the Browser
- Open Diagnostics tab on brave://rewards-internals
- EXPECTATION:
   * Diagnostics page has values
